### PR TITLE
Transfers: always skip ACTIVE transfers in receiver. #5335

### PR DESCRIPTION
**Pull request by rcarpa**

### CODE REVIEW

**Comment by bari12** [comment]
Any idea why the test-cases don't run here?

**Comment by joeldierkes** [comment]
Could be a problem with GitHub again (like we had some weeks ago), however, there is no incident in the GitHub Status page.
We changed nothing in the 1.26-LTS pipeline, so I don't think this is on our side.
Maybe try to update the branch or create another PR?

File: lib/rucio/daemons/conveyor/receiver.py [hunk 1]
```diff
@@ -87,9 +87,7 @@ def on_message(self, frame):
            and 'issuer' in msg['job_metadata'].keys() \
            and str(msg['job_metadata']['issuer']) == str('rucio'):
 
-            if 'job_state' in msg.keys() and (
-                    str(msg['job_state']) != str('ACTIVE')
-                    or str(msg['job_state']) == str('ACTIVE') and 'job_m_replica' in msg.keys() and (str(msg['job_m_replica']).lower() == str('true'))):
+            if 'job_state' in msg.keys() and str(msg['job_state']) != str('ACTIVE'):
```
Comment (joeldierkes) [inline lib/rucio/daemons/conveyor/receiver.py:90]: This seems oddly written, I'd suggest:

```python
if 'job_state' in msg and msg['job_state'] != 'ACTIVE':
```

- The python docu suggst to use `key in dict` to check if a key is contained in a dictionary (http://docs.python.org/3/tutorial/datastructures.html#dictionaries)
- `'ACTIVE'` is a string already, the cast is redundant
- `msg['job_state']` should always be a string if I'm not mistaken. It makes sense to throw an error if that is not the case so we can investigate that. The cast to a string and comparison afterwards would not throw an error...
  We maybe have to test if the value is `NONE` as a special case, but I'd rather test that explicitly

**Review by joeldierkes — CHANGES_REQUESTED** [review]

File: lib/rucio/daemons/conveyor/receiver.py (hunk 1 above)
Comment (bari12) [reply lib/rucio/daemons/conveyor/receiver.py:90]: It should still work as it is. @rcarpa can check this on return, I need to merge it in the release now.

**Pull request merged**